### PR TITLE
List tests refactoring

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by ksercs — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 1]
```diff
@@ -1340,46 +1340,65 @@ QUnit.module('options changed', moduleSetup, () => {
         assert.strictEqual($list.find(`.${LIST_SELECT_ALL_LABEL_CLASS}`).text(), 'Select All');
     });
 
-    QUnit.test('menuItems', function(assert) {
-        const actionSpy = sinon.spy();
-        const menuItems = [{ text: 'action', action: actionSpy }];
-
+    QUnit.test('menuItems can be added runtime', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1343]: ... at runtime

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 2]
```diff
@@ -1340,46 +1340,65 @@ QUnit.module('options changed', moduleSetup, () => {
...
+        const $item = $list.find('.dx-list-item').eq(0);
+        const contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
         $item.trigger(contextMenuEvent);
-        $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+
+        const $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
         assert.strictEqual($menuItems.length, 0, 'items count is correct');
+    });
 
+    QUnit.test('menuItems can be changed to empty array and then back', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1381]: It looks like this test duplicates the previous two ones. Do we need it?

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 3]
```diff
@@ -1340,46 +1340,65 @@ QUnit.module('options changed', moduleSetup, () => {
...
-        $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+
+        const $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
         assert.strictEqual($menuItems.length, 1, 'items count is correct');
-        $menuItems.eq(0).trigger('dxclick');
-        assert.strictEqual(actionSpy.callCount, 2, 'action is called twise');
     });
 
-    QUnit.test('menuMode', function(assert) {
+    QUnit.test('menuMode can be changed from context to slide runtime', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1401]: ... at runtime'

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 4]
```diff
@@ -1388,65 +1407,123 @@ QUnit.module('options changed', moduleSetup, () => {
...
         pointer.start().swipeStart().swipe(-0.5).swipeEnd(-1, -0.5);
+        const $actionButtons = $item.find('.dx-list-slide-menu-button');
 
-        let $actionButtons = $item.find('.dx-list-slide-menu-button');
         assert.strictEqual($actionButtons.length, 1, 'items count is correct');
-        $actionButtons.eq(0).trigger('dxclick');
-        assert.strictEqual(actionSpy.callCount, 1, 'action is called once');
+    });
+
+    QUnit.test('menuMode can be changed from slide to context runtime', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1421]: '... at runtime'

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 5]
```diff
@@ -1388,65 +1407,123 @@ QUnit.module('options changed', moduleSetup, () => {
...
+        const $item = $list.find('.dx-list-item').eq(0);
         const contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
         $item.trigger(contextMenuEvent);
         const $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+
         assert.strictEqual($menuItems.length, 1, 'items count is correct');
     });
 
-    QUnit.test('useInkRipple', function(assert) {
+    QUnit.test('menuMode can be changed from slide to context runtime and then back', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1441]: It looks like this test duplicates the previous two ones. Do we need it?

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 6]
```diff
@@ -1388,65 +1407,123 @@ QUnit.module('options changed', moduleSetup, () => {
...
-        inkRippleShowingWave = $item.find(toSelector(INKRIPPLE_WAVE_SHOWING_CLASS));
+        const inkRippleShowingWave = $item.find(toSelector(INKRIPPLE_WAVE_SHOWING_CLASS));
+
         assert.strictEqual(inkRippleShowingWave.length, 1, 'inkripple feedback works');
+
         pointer.start('touch').up();
+        clock.restore();
+    });
+
+    QUnit.test('useInkRipple can be changed to false and then back to true at runtime', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1508]: It looks like this test duplicates the previous two ones. Do we need it?

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js (hunk 2 above)
Comment (alexander-kotov-dx) [reply testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1381]: I think yes, we do. It is rare, but i remember some issues in similar cases

Comment (San4es) [reply testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:1381]: > i remember some issues in similar cases

Any examples?

**Pull request merged**